dts/arm/st/stm32f7.dtsi

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2018 Yurii Hamann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <st/mem.h>
+#include <dt-bindings/clock/stm32_clock.h>
+#include <dt-bindings/gpio/gpio.h>
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m7";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@20000000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x20000000 DT_SRAM_SIZE>;
+	};
+
+	flash0: flash@8000000 {
+	compatible = "soc-nv-flash";
+	label = "FLASH_STM32";
+	reg = <0x08000000 DT_FLASH_SIZE>;
+	write-block-size = <1>;
+	};
+
+	soc {
+		rcc: rcc@40023800 {
+			compatible = "st,stm32-rcc";
+			clocks-controller;
+			#clock-cells = <2>;
+			reg = <0x40023800 0x400>;
+			label = "STM32_CLK_RCC";
+		};
+
+		gpioa: gpio@40020000 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x40020000 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x00000001>;
+				label = "GPIOA";
+		};
+
```

Comment:
The indentation is also wrong here, compared to the rcc one.

Reply:
Is there a reason why the all the gpio entries are not included in pinctrl entry like on other STM32 SoCs? I guess that it also explains the indentation difference here.

Reply:
I am planning to add pinctrl in next PR. STM32 boards still use pinconf array instead of device tree definitions.

Reply:
AFAIK, no boards today uses device tree definitions. There is some work on going, but will take some time (pinctrl API introduction then device tree based generation).
Meanwhile, indeed, you'd better move gpio nodes under pinctrl node, like other STM32 SoCs

Reply:
OK.